### ISSUE TITLE
First tutorial "Preflight" bug

### ISSUE THREAD

**Issue by wlbragg** [Juanvvc/c172p-detailed#849]
See https://forum.flightgear.org/viewtopic.php?f=25&t=30813&p=297493#p297493

Basically I think the trouble is when the tutorial exits the aircraft it does so in the cockpit view and it appears you cannot interact with the external items when in cockpit view.
Verified.

**Comment by gilbertohasnofb** [comment]
I think @onox and I had a discussion once about the cockpit view and we decided that it was more realistic to forbid the user to click on objects outside the cockpit. It also helped to remove some of the yellow contours that appear with Ctrl+C, and it looked cleaner. I would not be very happy in reverting that, I would rather ask that we take a look on how the tutorials work and try to change the view for an external view when doing those things.

I don't have a chance to test it now, but wasn't the tutorial done with the walker? If so, I believe that would be an external view and that would allow the pitot tube cover to be removed.

**Comment by wlbragg** [comment]
>try to change the view for an external view when doing those things.

I think that may be all that is needed.

>wasn't the tutorial done with the walker? If so, I believe that would be an external view and that would allow the pitot tube cover to be removed.

It actually says it is still in the cockpit view when you leave the cockpit, even though it does look like a walker view.

**Comment by gilbertohasnofb** [comment]
> It actually says it is still in the cockpit view when you leave the cockpit, even though it does look like a walker view.

Every view has a parameter that says if it is internal or external, and this value is output as a boolean via the property `/sim/current-view/internal`. Look for instance this declaration the file `c172p-views.xml`:

```
    <view n="102">
        <name>Cockpit Right Seat View</name>
        <type>lookfrom</type>
        <internal type="bool">true</internal>
```

And I see now that the walker view, defined at `FGDATA/Aircraft/Generic/Human/Include/walker-include.xml`, does not have this line about internal type:

```		<view n="110">
			<name>Walk View</name>
			<type>lookfrom</type>
			<enabled type="bool">false</enabled>
			<config>
				<default-field-of-view-deg type="double">55.0</default-field-of-view-deg>
				<eye-alt-ft-path>sim/walker/altitude-ft</eye-alt-ft-path>
				<eye-heading-deg-path>sim/walker/heading-deg</eye-heading-deg-path>
				<eye-lat-deg-path>sim/walker/latitude-deg</eye-lat-deg-path>
				<eye-lon-deg-path>sim/walker/longitude-deg</eye-lon-deg-path>
				<eye-pitch-deg-path>sim/walker/pitch-deg</eye-pitch-deg-path>
				<eye-roll-deg-path>sim/walker/roll-deg</eye-roll-deg-path>
				<ground-level-nearplane-m type="double">0.5</ground-level-nearplane-m>
				<x-offset-m type="double">0.6</x-offset-m>
				<y-offset-m type="double">2.1</y-offset-m>
				<z-offset-m type="double">9.4</z-offset-m>
				<limits>
					<enabled type="bool">false</enabled>
				</limits>
			</config>
		</view>
```

And nor does the "walker orbit view".

Maybe someone could add the line `<internal type="bool">false</internal>` to the walker view in that file and make a test, and if that solves it then we will have to ask the devs in the mailing list for a fix.

I can't test this now, but maybe in the evening I will have time if no one has done it before.

**Comment by gilbertohasnofb** [comment]
Hmm, I just tested the walker in the sim but that's not the reason of the problem, the walker is not recognized as an internal view.

**Comment by gilbertohasnofb** [comment]
I tested the tutorial and this is the reason for this bug: the tutorial does not use the walker, it does use the cockpit view all the time, including when the views are outside! So either we find a way around this by using different views or we will have to revert that commit that makes external objects non clickeable from the cockpit (pitot tube cover, tie-downs, oil level, chocks, etc.)

@onox @dg-505 do you guys have some idea how to tackle this?

**Comment by gilbertohasnofb** [comment]
@wlbragg if you can, please reply to that thread in the forum letting those users know the bug has been confirmed and that we will work of a solution soon

**Comment by dg-505** [comment]
Thanks for letting me know. I'll take a look when I'm home from work today.

**Comment by wlbragg** [comment]
👍

**Comment by gilbertohasnofb** [comment]
@dg-505 great, thanks a lot! Let me know if I can be of any help

**Comment by dg-505** [comment]
I'm on it. Turned out to be a little bit more difficult as expected :roll_eyes: 
I think I'll get it finished this evening

**Comment by dg-505** [comment]
Fixed in #858
